include/IrcCore/ircdebug_p.h

```diff
@@ -142,8 +145,19 @@ static bool irc_debug_enabled(IrcConnection* c, uint l)
 
         dbg_init = true;
     }
-
-    return l <= dbg_level && (dbg_name.isEmpty() || QRegularExpression(QRegularExpression::wildcardToRegularExpression(dbg_name), QRegularExpression::CaseInsensitiveOption).match(c->displayName()).hasMatch());
+    if (l <= dbg_level ) {
+        if (!dbg_name.isEmpty()) {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 0)
+            return QRegularExpression(QRegularExpression::wildcardToRegularExpression(dbg_name),
+                       QRegularExpression::CaseInsensitiveOption).match(c->displayName()).hasMatch();
+#else
+            return QRegExp(dbg_name,
+                   Qt::CaseInsensitive,
+                   QRegExp::Wildcard).exactMatch(c->displayName());
+#endif
+        }
+    }
+    return false;
```

Comment:
These are the conditions:
- the debug level must be enabled, and
- either the name must be either empty, or it must match the filter

Before:
```
return level_ok && (is_empty || regexp_match);
```

After:
```
if (level_ok) {
    if (!is_empty) {
      return regexp_match;
    }
}
return false;
```

Which is the same as:
```
return level_ok && !is_empty && regexp_match;
```

Before, an empty name was accepted as long as the level was ok. After, an empty name would no longer be accepted.

Reply:
I fixed that now, I think had trouble reading that before.